internal/errors/error.go

```diff
@@ -109,7 +161,7 @@ func Wrap(e error, op Op, opt ...Option) error {
 		opt = append(opt, WithWrap(e))
 	}
 
-	return E(opt...)
+	return EDeprecated(opt...)
```

Comment:
Weird I came to edit my comment but couldn't find it...
I am not sure we should be calling `EDeprecated` from `Wrap` or `Convert` this would block us from eventually deprecating it.  I would suggest we add a new function `e` that performs what `EDeprecated` currently does.   Wrap and Convert can then call `e` instead of `EDeprecated`, `EDeprecated` can just return `e` and `NewDeprecated` can do its thing then call `e`.  This also removes the duplicate code between the `deprecated` functions and the new ones, it also means we could update all tests to just test `e` for now

Reply:
I've refactored a bit to resolve this issue.